### DISCUSSION TITLE
problem when using raplicatedReplacingMergeTree not work after a time

### DISCUSSION THREAD

**Discussion by lucasenabit** [ClickHouse/ClickHouse#55429] [Q&A]
I have a big problem in my project, I really need a solution. A brief description of the scenario: My environment cannot contain duplicate data, this needs to be guaranteed at least on the current day's partition. This way, I used the replicatedReplacingMergeTree engine, at first it was working, but the next day it stopped doing the work and remained like that for the next 3 days, where I was forced to run optimize table final deduplicate by xxx , the execution was relatively fast , and in the end removed the duplications. However, I can't run this all the time in production, replacing should work and it isn't working.

Environment with an average of 16 million records/day

Please, if anyone has a solution for this case I will be very grateful, I am currently testing with view materialized to try to get around the problem, but still no hope.

Below is the table structure used.

ENGINE = ReplicatedReplacingMergeTree('/clickhouse/tables/alma/{shard}/transaction_ori_2', '{replica}')
  PARTITION BY toYYYYMMDD(dataTransacao_Bit07)
  Primary Key (dataTransacao_Bit07, flag_id, modeEntradaBit22, modeEntradaBit22_id, productBit48, productBit48_id, accreditorBit32, accreditorBit32_id, establishmentOperacao_Bit43, funcaoBit24_id, codigoRespostaBit39_id, codigoRespostaBit39, funcaoBit 24,emitter,emitter_id,bin6,bin8,marcasAlerts_Bit48,
              brandDDN_Bit48,Cnpj_Bit48)
  order it24,emitter,emitter_id,bin6,bin8,marcasAlertas_Bit48,
              brandDDN_Bit48,Cnpj_Bit48,
           establishmentName_Bit43,Mid_Bit42,pan_Bit02,productBin,id);

distributed table

CREATE TABLE bam.transaction_2 on CLUSTER `alma`
as bam.transaction_ori_2
ENGINE = Distributed(`alma`, `bam`, `transaction_ori_2`, if( site = 'SP' , 1, 0));

**Comment by den-crane** [comment]
everything here is expected.

https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/replacingmergetree

_Data deduplication occurs only during a merge. Merging occurs in the background at an unknown time, so you can’t plan for it. Some of the data may remain unprocessed._

https://clickhouse.com/docs/en/sql-reference/statements/select/from#final-modifier

https://kb.altinity.com/engines/mergetree-table-engine-family/replacingmergetree/altinity-kb-replacingmergetree-does-not-collapse-duplicates/

https://kb.altinity.com/engines/mergetree-table-engine-family/replacingmergetree/

https://kb.altinity.com/altinity-kb-schema-design/row-level-deduplication/

**Reply by lhbsaldana** [reply]
Have you tried querying the distributed table with the `FINAL` keyword? This forces the merge to be happen at query time.
```
SELECT * FROM distributed_table FINAL;  
```
